### ISSUE TITLE
unrelated subscriptions break previously defined observable streams

### ISSUE THREAD

**Issue by mike-co** [ReactiveX/rxjs#2370]
RXJS version 5.0.3
```
Rx.Observable.of(1).switchMap(()=> {
  console.log('hi switchMap');
  return Rx.Observable.interval(1000).take(1);
})
.do(()=> {console.log('hi do')})
.finally(() => console.log('hi finally'))
.subscribe();

Rx.Observable.of(0).subscribe();
```

the expected output is
> "hi switchMap"
> "hi do"
> "hi finally" 

the actual output is
> "hi switchMap"
> "hi finally"

However, if we have `Rx.Observable.of(0).subscribe(()=>{});` with the empty function inside then the expected output is achieved.

codepen.io link with issue: http://codepen.io/codepen-michael/pen/GrzrqB

**Comment by trxcllnt** [comment]
Yes, this will be fixed when #2238 is merged

**Comment by jayphelps** [comment]
Hey @mike-co. Thank you for reporting--especially including a codepen (!!!) As mentioned, this will be fixed fairly soon. There is also another issue ticket we're using to track this. #2237 if you have additional details, feel free to add them there.
